src/Containers/Microsoft.NET.Build.Containers/ImageBuilder.cs

```diff
@@ -32,6 +37,10 @@ internal ImageBuilder(ManifestV2 manifest, ImageConfig baseImageConfig)
     /// </summary>
     internal BuiltImage Build()
     {
+        // before we build, we need to make sure that any image customizations occur
+        AssignUserFromEnvironment();
+        AssignPortsFromEnvironment();
```

Comment:
Here's the change @vlada-shubina requested - we assign these values immediately before construction, so that no matter how you `.Build()` an Image, you get the value applied. This means we don't need code in both ContainerBuilder and CreateNewImage